content/docs/tasks/telemetry/kiali/index.md

```diff
@@ -0,0 +1,363 @@
+the mesh as well as your Istio configuration objects. You will also use the
+Kiali Public API to generate graph data in the form of consumable JSON.
+
+The [Bookinfo](/docs/examples/bookinfo/) sample application is used as the
+example application throughout this task.
+
+## Before you begin
+
+You first need to create a secret in your Istio namespace with credentials that
+users will use to authenticate to Kiali. See the
+[Helm README](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/README.md#installing-the-chart)
+for details, but here is a command you can run to create a secret:
+
+```bash
+USERNAME=$(echo -n 'admin' | base64)
+PASSPHRASE=$(echo -n 'mysecret' | base64)
+NAMESPACE=istio-system
+kubectl create namespace $NAMESPACE
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kiali
+  namespace: $NAMESPACE
+  labels:
+    app: kiali
+type: Opaque
+data:
+  username: $USERNAME
+  passphrase: $PASSPHRASE
+EOF
+```
+
+Once a Kiali secret is created, you can now install Kiali via Helm by following
+[these instructions](/docs/setup/kubernetes/helm-install/). Note that you need
+to use the `--set kiali.enabled=true` option when running the `helm` command.
+For example:
+
+{{< text bash >}}
+$ helm template --set kiali.enabled=true install/kubernetes/helm/istio --name istio --namespace istio-system > $HOME/istio.yaml
+$ kubectl apply -f $HOME/istio.yaml
+{{</ text >}}
+
+Once Istio and Kiali are installed, you should then deploy an application such
+as the [Bookinfo](/docs/examples/bookinfo/) sample application.
+
+{{< idea_icon >}} The above installation instructions assume you have `helm`
+and want to use it to install Kiali. Alternatively, you can install Kiali by
+following the [Kiali install instructions here](https://www.kiali.io/gettingstarted/).
+
+## Generating a Service Graph
+
+1.  Verify that the service is running in your cluster.
+
+    In Kubernetes environments, execute the following command:
+
+    {{< text bash >}}
+    $ kubectl -n istio-system get svc kiali
+    NAME           CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+    kiali          10.59.253.165   <none>        20001/TCP  30s
+    {{< /text >}}
+
+1.  Send traffic to the mesh.
+
+    For the Bookinfo sample, [follow these instructions to determine its
+    GATEWAY_URL](/docs/examples/bookinfo/#determining-the-ingress-ip-and-port)
+    and then visit `http://$GATEWAY_URL/productpage` in your web browser or
+    issue the following command:
+
+    {{< text bash >}}
+    $ curl http://$GATEWAY_URL/productpage
+    {{< /text >}}
+
+    Refresh the page a few times, or send the command a few times, to generate a small amount of traffic.
+    To continually send requests, you can use the `watch` command if available on your system:
+
+    {{< text bash >}}
+    $ watch -n 1 curl -o /dev/null -s -w %{http_code} $GATEWAY_URL/productpage
+    {{< /text >}}
+
+1.  Login to the Kiali UI.
```

Comment:
Break this out into two separate steps:
Determine the Kiali URL
Log into the console.

Reply:
This is done